x-pack/solutions/security/plugins/security_solution/public/asset_inventory/components/onboarding/hover_for_explanation.tsx

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React from 'react';
+import { EuiToolTip, useEuiTheme } from '@elastic/eui';
+import { css } from '@emotion/react';
+
+/**
+ * A component that adds a tooltip to its children and adds an underline to indicate that
+ * the children can be hovered for more information.
+ */
+export const HoverForExplanation = ({
+  children,
+  tooltipContent,
+}: {
+  children: React.ReactNode;
+  tooltipContent: React.ReactNode;
+}) => {
```

Comment:
cool component!
maybe we can reuse EuiToolTipProps instead of creating new ones
```suggestion
export const HoverForExplanation = ({
  children,
  tooltipContent,
}: EuiToolTipProps) => {
```

`EuiToolTipProps['children']` and `EuiToolTipProps['tooltipContent']` only if you'd prefer

Reply:
nice suggestion, it actually made me think about improving the component to be a High Order Component of the EuiTooltip leaving room for passing all the `EuiTooltip` props